test/test-utils/matchers/to-have-fainted-matcher.ts

```diff
@@ -0,0 +1,31 @@
+import { getPokemonNameWithAffix } from "#app/messages";
+import { isPokemonInstance, receivedStr } from "#test/test-utils/test-utils";
+import type { MatcherState, SyncExpectationResult } from "@vitest/expect";
+
+/**
+ * Matcher to check if a Pokemon has fainted
+ * @param received - The object to check. Should be a {@linkcode Pokemon}.
+ * @returns Whether the matcher passed
+ */
+export function toHaveFaintedMatcher(this: MatcherState, received: unknown): SyncExpectationResult {
+  if (!isPokemonInstance(received)) {
+    return {
+      pass: this.isNot,
+      message: () => `Expected Pokemon, but got ${receivedStr(received)}!`,
+    };
+  }
+
+  const { hp } = received;
+  const maxHp = received.getMaxHp();
+  const pass = received.isFainted() === true && hp === 0;
```

Comment:
```suggestion
  const pass = received.isFainted() === true && hp === 0;
```
`Pokemon#isFainted` is already checking for the pokemon having 0 HP.